### PR TITLE
feat: Validate a channel before adding client to list

### LINKED ISSUE #113: Validate a channel before adding client to list
When someone opens the chat link, we make a WebSocket connection and maintain a list of connected clients. We should only care about clients on a valid chat link.

https://github.com/muke1908/chat-e2ee/blob/e08a23673a24cd41967c9d2d4dadb4ed1b39f0bf/backend/socket.io/listeners.js#L4

It shouldn't add user to list if the channel is invalid. Validate using channelValid function 
https://github.com/muke1908/chat-e2ee/blob/e08a23673a24cd41967c9d2d4dadb4ed1b39f0bf/backend/api/chatLink/utils/validateChannel.js#L4

### PR DESCRIPTION
**Pull request by drmartell**
Resolves #113 by adding using async/await and try...catch

This is my first open source contribution so open to any feedback.

### CODE REVIEW

File: backend/socket.io/listeners.js [hunk 1]
```diff
@@ -1,8 +1,17 @@
 const clients = require('./clients');
+const channelValid = require('../api/chatLink/utils/validateChannel');
 
 const connectionListener = (socket, io) => {
-  socket.on('chat-join', (data) => {
+  socket.on('chat-join', async (data) => {
     const { userID, channelID, publicKey } = data;
+    try {
+      await channelValid(channelID);
```
Comment (muke1908) [inline backend/socket.io/listeners.js:8]: The [function](https://github.com/muke1908/chat-e2ee/blob/e08a23673a24cd41967c9d2d4dadb4ed1b39f0bf/backend/api/chatLink/utils/validateChannel.js#L4) `channelValid` returns an object like
```
return {
    valid: <true|false>,
    state: <state>
};
```
So it should be something like:
```
const { valid } = await channelValid(channelID);
if(!valid) {
    return
}
```

For reference, you can check this implementation 
https://github.com/muke1908/chat-e2ee/blob/e08a23673a24cd41967c9d2d4dadb4ed1b39f0bf/backend/api/messaging/index.js#L22

**Review by muke1908 — CHANGES_REQUESTED** [review]

**Comment by muke1908** [comment]
@drmartell thank you for contributing!!

**Comment by muke1908** [comment]
@all-contributors please add @drmartell for code

File: backend/socket.io/listeners.js (hunk 1 above)
Comment (drmartell) [reply backend/socket.io/listeners.js]: yes, of course, thank you!

**Pull request merged**